src/util/logging/hooks.go

```diff
@@ -0,0 +1,84 @@
+package logging
+
+import (
+	"reflect"
+
+	"github.com/sirupsen/logrus"
+)
+
+func NewReplayHook(logger *Logger) ReplayHook {
+	return ReplayHook{
+		Logger: logger,
+		excludeTypes: map[reflect.Type]int{
+			reflect.TypeOf(ReplayHook{}):    1,
+			reflect.TypeOf(ModuleLogHook{}): 1,
+		},
+	}
+}
+
+// Do not replay hooks of given exclude types
+func ExclusiveReplayHook(logger *Logger, exclude []reflect.Type) (h ReplayHook) {
+	h = NewReplayHook(logger)
+	h.excludeTypes = nil
+	for _, _type := range exclude {
+		h.excludeTypes[_type] = 1
+	}
+	return
+}
+
+// Hook for replaying hooks bound to another logger
+type ReplayHook struct {
+	Logger *Logger
+	// FIXME : Go lang sets?
+	excludeTypes map[reflect.Type]int
```

Comment:
The convention for sets is to map to the empty struct, e.g.

`excludeTypes map[reflect.Type]struct{}`

To add an element, you would do `excludeTypes[t] = struct{}{}`